shell/browser/ui/file_dialog_linux.cc

```diff
@@ -31,15 +31,15 @@ DialogSettings::~DialogSettings() = default;
 namespace {
 
 ui::SelectFileDialog::Type GetDialogType(int properties) {
-  if (properties & OPEN_DIALOG_OPEN_FILE)
+  if (properties & OPEN_DIALOG_OPEN_FILE) {
+    if (properties & OPEN_DIALOG_MULTI_SELECTIONS)
+      return ui::SelectFileDialog::SELECT_OPEN_MULTI_FILE;
     return ui::SelectFileDialog::SELECT_OPEN_FILE;
+  }
 
   if (properties & OPEN_DIALOG_OPEN_DIRECTORY)
     return ui::SelectFileDialog::SELECT_FOLDER;
 
-  if (properties & OPEN_DIALOG_MULTI_SELECTIONS)
-    return ui::SelectFileDialog::SELECT_OPEN_MULTI_FILE;
-
   return ui::SelectFileDialog::SELECT_OPEN_FILE;
```

Comment:
These conditionals + `SELECT_OPEN_FILE` being returned from two different logic branches is really hard to follow.

Would this be equivalent?

```c++
if (properties & OPEN_DIALOG_OPEN_DIRECTORY)
  return ui::SelectFileDialog::SELECT_FOLDER;

if (properties & OPEN_DIALOG_MULTI_SELECTIONS)
  return ui::SelectFileDialog::SELECT_OPEN_MULTI_FILE;

return ui::SelectFileDialog::SELECT_OPEN_FILE;
```